compiler/src/dmd/declaration.d

```diff
@@ -39,32 +40,21 @@ import dmd.common.outbuffer;
 import dmd.rootobject;
 import dmd.root.filename;
 import dmd.target;
+import dmd.targetcompiler;
 import dmd.tokens;
 import dmd.typesem : typeSemantic, size;
 import dmd.visitor;
 
-version (IN_GCC) {}
-else version (IN_LLVM) {}
-else version = MARS;
 
 /******************************************
  */
 void ObjectNotFound(Loc loc, Identifier id)
 {
     global.gag = 0; // never gag the fatal error
-    error(loc, "`%s` not found. object.d may be incorrectly installed or corrupt.", id.toChars());
-    version (IN_LLVM)
-    {
-        errorSupplemental(loc, "ldc2 might not be correctly installed.");
-        errorSupplemental(loc, "Please check your ldc2.conf configuration file.");
-        errorSupplemental(loc, "Installation instructions can be found at http://wiki.dlang.org/LDC.");
-    }
-    else version (MARS)
-    {
-        errorSupplemental(loc, "dmd might not be correctly installed. Run 'dmd -man' for installation instructions.");
-        const dmdConfFile = global.inifilename.length ? FileName.canonicalName(global.inifilename) : "not found";
-        errorSupplemental(loc, "config file: %.*s", cast(int)dmdConfFile.length, dmdConfFile.ptr);
-    }
+    const dmdConfFile = global.inifilename.length ? FileName.canonicalName(global.inifilename) : "not found";
```

Comment:
This should work with LDC too, but I think it'd be better to let the `hostObjectNotFound` infer any config files it wants to print out.

Reply:
This is refactoring, not a redesign of what it does.